### PR TITLE
SSH Brute Force Analysis

### PR DESCRIPTION
**Pull request by roshanmaskey**
This PR adds `SSHBruteForcePlugin` (`ssh_bruteforce.py`) that analysis SSH authentication events and provides the summary of brute force analysis report and automatically tags the relevant events. 

**SSHBruteForcePluing Result**

The figure below shows the result of the analyzer.

sshbruteforceplugin_result

**Labeled Events**

The figure below shows relevant SSH events labeled and stared.

sshbruteforceplugin_label

**Checks**

- [x] All tests succeed.
- [x] Unit tests added.
- [ ] e2e tests added.
- [ ] Documentation updated.

### CODE REVIEW

**Comment by berggren** [comment]
Ack, I will review asap.

**Comment by berggren** [comment]
Hey @roshanmaskey good stuff! I started looking at this today, and I have a couple of higher level thoughts.

- Maybe consider refactor to a proper plugin system for authentication analyzers (see the chain analyzers for example)
- The use of pyparsing - I'm wondering if we already have some of this data extracted via plaso? And if not, should we add it there?

I'll set up a sync for us together with @jkppr to progress this asap.

**Comment by berggren** [comment]
Adding @jkppr to review from the analysis side

**Review by jkppr — COMMENTED** [review]
As discussed offline: This PR will get some more work by @roshanmaskey with new ideas and approaches from the last month.
Main points are:
* Update the demo data to be compatible with the newest Timesketch/Plaso version
* If possible make use of the `syslog` plaso parser to extract the necessary fields. This would allow to skip the usage of pyparsing.
* If field  parsing is necessary, check out the feature extractor function to take care of this before running your analyzer.

**Comment by roshanmaskey** [comment]
@jkppr ready for review.

File: timesketch/lib/analyzers/auth.py
Comment (jkppr) [inline timesketch/lib/analyzers/auth.py:1]: Let's discuss in our next 1:1 the pro & cons of building this as an analyzer plugin system.

File: timesketch/lib/analyzers/ssh_bruteforce_test.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
```
Comment (jkppr) [inline timesketch/lib/analyzers/ssh_bruteforce_test.py:1]: Not needed in python 3 anymore.

File: timesketch/lib/analyzers/ssh_bruteforce.py [hunk 2]
```diff
@@ -0,0 +1,388 @@
+# -*- coding: utf-8 -*-
```
Comment (jkppr) [inline timesketch/lib/analyzers/ssh_bruteforce.py:1]: Not needed in python 3 anymore.

**Review by jkppr — CHANGES_REQUESTED** [review]
First round of comments.
I need a bit more time to think about the usage of pyparsing and if we could archive the same goal using the feature extractor. I will come back to this in round 2 then :)

**Pull request closed without merging**